pcsx2/GS/Renderers/HW/GSTextureCache.cpp

```diff
@@ -2213,21 +2206,55 @@ void GSTextureCache::Target::Update()
 
 	// m_renderer->m_perfmon.Put(GSPerfMon::Unswizzle, w * h * 4);
 
-	// Copy the new GS memory content into the destination texture.
-	if (m_type == RenderTarget)
+	// The idea: If we have more than one dirty rectangle, we upload the whole thing, and then
+	// copy/upscale the specific areas we want into it.
+	if (m_dirty.size() > 1)
 	{
-		GL_INS("ERROR: Update RenderTarget 0x%x bw:%d (%d,%d => %d,%d)", m_TEX0.TBP0, m_TEX0.TBW, r.x, r.y, r.z, r.w);
+		const GSVector2i bs = GSLocalMemory::m_psm[m_TEX0.PSM].bs;
+		for (const auto& dr : m_dirty)
```

Comment:
I like this.
This is also kind of part of #5534 (https://github.com/PCSX2/pcsx2/pull/5534/files#diff-a08fccf34ca3d5af6f2ba46d887e2f60833415531482f5ac08a9f4848b9f8f45R1758).